### ISSUE TITLE
buttonConfiguration property editor wont list row actions when used by custom component.

### ISSUE THREAD

**Issue by poirazis** [Budibase/budibase#15116]
When used by a custom component, the butotonConfiguration property editor will not list the available Row Actions to be selected as buttons.

**Comment by aptkingston** [comment]
Hey @poirazis. This is actually by design.

The row action templates deliberately only work for a small whitelist of components, namely the form block and table. That's because the configuration for the row action requires a row ID binding for the row to act on, and that's different depending on what component is using it. We know what this binding is for form blocks and tables, so those are the cases we support.

I'm afraid I don't think it will be possible to open this up to custom components as I don't think there's a clean way of adding a setting to basically mean "how do you extract the row ID binding from the component these buttons are used inside". However there's nothing stopping you from fetching row actions yourself (`API.rowActions.fetch(tableId)`) and rendering something more opinionated, if that helps.

**Comment by poirazis** [comment]
Hey @aptkingston 
will a different property editor be provided for custom components to let users pick from the available row actions ( yes , appropriate binding / enrichment will be done by the component ) ? 
an let say , rowActionConfiguration, that will be used independantly just for selection of row actions ?

**Comment by aptkingston** [comment]
I don't think so, only because it wouldn't save the user much time. It would only save you 1 or 2 clicks - since it's basically the same as adding a normal button, giving it a name, then adding the "row action" button action. You can still use row actions from custom components, but it just takes a few more clicks.
